src/environmentd/src/http.rs

```diff
@@ -368,6 +374,31 @@ async fn auth<B>(
             .ok_or(AuthError::InvalidCertUserName)?,
     };
 
+    let user = auth(frontegg, tls_mode, cert_user, req.headers()).await?;
+
+    // Add the authenticated user as an extension so downstream handlers can
+    // inspect it if necessary.
+    req.extensions_mut().insert(user);
+
+    // Run the request.
+    Ok(next.run(req).await)
+}
+
+async fn auth(
+    frontegg: &Option<FronteggAuthentication>,
+    tls_mode: Option<TlsMode>,
+    mut user: Option<String>,
+    headers: &HeaderMap<HeaderValue>,
+) -> Result<AuthedUser, AuthError> {
```

Comment:
This function split seems funky. Seems like `http_auth` ought to be responsible for pulling out the headers, and `auth` ought to take a new enum like:

```rust
enum Credentials {
    Password { username: String, password: String},
    Token { token: String },
}
```

Then constructing the call to `auth` in `init_ws` won't be so awkward.